daemon/workit.cpp

```diff
@@ -501,11 +522,13 @@ int work_it(CompileJob &j, unsigned int job_stat[], MsgChannel *client, CompileR
                     }
 
                     kill(pid, SIGTERM); // Most likely crashed anyway ...
-                    return_value = EXIT_COMPILER_CRASHED;
-                    client_fd = -1;
-                    input_complete = true;
+                    if (input_complete) {
+                        return_value = EXIT_COMPILER_CRASHED;
+                    }
                     delete fcmsg;
                     fcmsg = 0;
+                    close(sock_in[1]);
+                    sock_in[1] = -1;
```

Comment:
Instead of negative one as a hard coded number use `#define COMPILED_CRASH_SOCKET_CLOSED -1` someplace and reference that here.  I'd probably use sock_in[in] <0 as the check for error condition elsewhere.